### PR TITLE
correct macros.genSym comment doc

### PR DESCRIPTION
**Pull request by bung87**
refs: https://github.com/nim-lang/Nim/issues/17851

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -499,8 +499,10 @@ proc bindSym*(ident: string | NimNode, rule: BindSymRule = brClosed): NimNode {.
 
 proc genSym*(kind: NimSymKind = nskLet; ident = ""): NimNode {.
   magic: "NGenSym", noSideEffect.}
-  ## Generates a fresh symbol that is guaranteed to be unique. The symbol
-  ## needs to occur in a declaration context.
+  ## Generates a fresh symbol.
+  ## if `ident` is not empty, the generated symbol maybe ambiguous.
```
Comment (Araq) [inline lib/core/macros.nim:503]: No, it's really always unique, it's just that the string representation of the code might not reflect this:

```nim

    of opcGenSym:
      decodeBC(rkNode)
      let k = regs[rb].intVal
      let name = if regs[rc].node.strVal.len == 0: ":tmp"
                 else: regs[rc].node.strVal
      if k < 0 or k > ord(high(TSymKind)):
        internalError(c.config, c.debug[pc], "request to create symbol of invalid kind")
      var sym = newSym(k.TSymKind, getIdent(c.cache, name), nextSymId c.idgen, c.module.owner, c.debug[pc])
      incl(sym.flags, sfGenSym)
      regs[ra].node = newSymNode(sym)
      regs[ra].node.flags.incl nfIsRef
```

Comment (bung87) [reply lib/core/macros.nim:503]: the problem is the symbol id is unique but the ident id is not. which will cause ambiguous, I've seen serveral issue about macro users submited issues about this.

Comment (Araq) [reply lib/core/macros.nim:503]: So users need to understand symbols but your documentation patch makes it worse.

Comment (bung87) [reply lib/core/macros.nim:503]: okay then, someone may have good one in mind.

**Pull request closed without merging**